substratest/assets.py

```diff
@@ -204,25 +212,22 @@ class Meta:
         }
 
 
-@dataclasses.dataclass(frozen=True)
 class DataSample(_Asset):
     key: str
     owner: str
     data_manager_keys: typing.List[str]
 
 
-@dataclasses.dataclass(frozen=True)
-class ObjectiveDataset(_DataclassLoader):
-    dataset_key: str
-    data_sample_keys: typing.List[str]
+class ObjectiveDataset(_InternalStruct):
+    dataset_key: str = None
```

Comment:
Are you sure None is required here? This comment applies to all the default None values. :)

Reply:
I just checked, this isn't required for all the current `None` values, but for some of them it is. I'll remove the unneeded 🙂